wideq/dryer.py

```diff
@@ -0,0 +1,205 @@
+import enum
+from typing import Optional
+
+from .client import Device, _UNKNOWN
+
+
+class DryerState(enum.Enum):
+    """The state of the dryer device."""
+
+    COOLING = '@WM_STATE_COOLING_W'
+    END = '@WM_STATE_END_W'
+    ERROR = '@WM_STATE_ERROR_W'
+    DRYING = '@WM_STATE_DRYING_W'
+    INITIAL = '@WM_STATE_INITIAL_W'
+    OFF = '@WM_STATE_POWER_OFF_W'
+    PAUSE = '@WM_STATE_PAUSE_W'
+    RUNNING = '@WM_STATE_RUNNING_W'
+    SMART_DIAGNOSIS = '@WM_STATE_SMART_DIAGNOSIS_W'
+    WRINKLE_CARE = '@WM_STATE_WRINKLECARE_W'
+    UNKNOWN = _UNKNOWN
+
+
+class DryLevel(enum.Enum):
+    """Represents the dry level setting of the dryer."""
+
+    CUPBOARD = '@WM_DRY27_DRY_LEVEL_CUPBOARD_W'
+    DAMP = '@WM_DRY27_DRY_LEVEL_DAMP_W'
+    EXTRA = '@WM_DRY27_DRY_LEVEL_EXTRA_W'
+    IRON = '@WM_DRY27_DRY_LEVEL_IRON_W'
+    LESS = '@WM_DRY27_DRY_LEVEL_LESS_W'
+    MORE = '@WM_DRY27_DRY_LEVEL_MORE_W'
+    NORMAL = '@WM_DRY27_DRY_LEVEL_NORMAL_W'
+    OFF = '-'
+    VERY = '@WM_DRY27_DRY_LEVEL_VERY_W'
+    UNKNOWN = _UNKNOWN
+
+
+class DryerError(enum.Enum):
+    """A dryer error."""
+
+    ERROR_AE = '@WM_US_DRYER_ERROR_AE_W'
+    ERROR_CE1 = '@WM_US_DRYER_ERROR_CE1_W'
+    ERROR_DE4 = '@WM_WW_FL_ERROR_DE4_W'
+    ERROR_DOOR = '@WM_US_DRYER_ERROR_DE_W'
+    ERROR_DRAINMOTOR = '@WM_US_DRYER_ERROR_OE_W'
+    ERROR_EMPTYWATER = '@WM_US_DRYER_ERROR_EMPTYWATER_W'
+    ERROR_F1 = '@WM_US_DRYER_ERROR_F1_W'
+    ERROR_LE1 = '@WM_US_DRYER_ERROR_LE1_W'
+    ERROR_LE2 = '@WM_US_DRYER_ERROR_LE2_W'
+    ERROR_NOFILTER = '@WM_US_DRYER_ERROR_NOFILTER_W'
+    ERROR_NP = '@WM_US_DRYER_ERROR_NP_GAS_W'
+    ERROR_PS = '@WM_US_DRYER_ERROR_PS_W'
+    ERROR_TE1 = '@WM_US_DRYER_ERROR_TE1_W'
+    ERROR_TE2 = '@WM_US_DRYER_ERROR_TE2_W'
+    ERROR_TE5 = '@WM_US_DRYER_ERROR_TE5_W'
+    ERROR_TE6 = '@WM_US_DRYER_ERROR_TE6_W'
+    UNKNOWN = _UNKNOWN
+
+
+class TempControl(enum.Enum):
+    """Represents temperature control setting."""
+
+    OFF = '-'
+    ULTRA_LOW = '@WM_DRY27_TEMP_ULTRA_LOW_W'
+    LOW = '@WM_DRY27_TEMP_LOW_W'
+    MEDIUM = '@WM_DRY27_TEMP_MEDIUM_W'
+    MID_HIGH = '@WM_DRY27_TEMP_MID_HIGH_W'
+    HIGH = '@WM_DRY27_TEMP_HIGH_W'
+    UNKNOWN = _UNKNOWN
+
+
+class TimeDry(enum.Enum):
+    """Represents a timed dry setting."""
+
+    OFF = '-'
+    TWENTY = '20'
+    THIRTY = '30'
+    FOURTY = '40'
+    FIFTY = '50'
+    SIXTY = '60'
+    UNKNOWN = _UNKNOWN
+
+
```

Comment:
I only kept a poll method as controlling an LG dryer/washer requires you to be present even if remote start is enabled.  This device will just report it's status and various attributes and not perform any actions on the dryer.